source/simulator/assembly.cc

```diff
@@ -2780,11 +1123,13 @@ namespace aspect
 
               material_model->evaluate(scratch.face_material_model_inputs,
                                        scratch.face_material_model_outputs);
-//TODO: the following doesn't currently compile because the get_quadrature() call returns
+
+              // TODO: formulations?
+
```

Comment:
I Have update the TODO to the following:      

              // TODO: Currently we do not supply reference density values to Stokes face assemblers.
              // This seems acceptable for now, since the only face assemblers are the ones for the melt
              // assembly where one would not want to use the reference density anyway. In case the reference
              // density is ever needed those assemblers can also query the adiabatic conditions for the
              // reference density.

All in all this TODO shows a design decision we might need to think about at some point. Is it the responsibility of `Simulator::local_assemble_stokes_system` to provide every necessary information to the assemblers inside of the `scratch` object, or should it only provide the basic information, and in any case an assembler needs more information it is responsible for getting this information? For now  `Simulator::local_assemble_stokes_system` provides or manipulates the information it gives to the assemblers in order to ensure the approximations (and following this design we would need to introduce `scratch.mass_density_face` variables and fill them for the face assemblers).